ci/test_commit_message.py

```diff
@@ -25,21 +25,24 @@ def lint_commit(commit_id):
 
     # Check 1: Subject, body and DCO exist
     # Note that git does not allow for empty subjects
-    if len(re.split('\n\n|\r', message)) <= 2:
+    msg_list = re.split('\n\n|\r', message)
+    commit_subject = msg_list[0]
+    if len(msg_list) <= 2:
         print("Commit {} does not have a body.".format(sha_short))
         check = False
+    else:
+        # pop the subject and signature
+        msg_list.pop(0)
+        msg_list.pop()
 
     # Check 2: Subject length is less than about 50
-    if len(re.split('\n\n', message)[0]) > 54:
+    if len(commit_subject) > 54:
         print(
             "The subject of commit {} should be 50 characters or less.".format(
                 sha_short))
         check = False
 
     # Check 3: Each line of the body is less than 72
-    msg_list = re.split('\n\n|\r', message)
-    msg_list.pop(0)
-    msg_list.pop()
     for msg in msg_list:
```

Comment:
This is a corner case scenario, but if we have a commit that is only a message subject and signature, since we wouldn't pop the subject and signature above, this is going to check that the signature line is less than 72 characters, which is not a requirement.

Reply:
I've already popped subject and signature in lines 35-36 of this file. So this corner case won't fail. The subject is stored in a variable (before popping) which will be checked for length <= 54 but the Signature won't ever be checked.

`if len(msg_list) <= 2:`
        `print("Commit {} does not have a body.".format(sha_short))`
        `check = False`
`else:`
       `# pop the subject and signature`
        `msg_list.pop(0)`
        `msg_list.pop()`

Reply:
We only pop the subject and signature if `if len(msg_list) <= 2` is false. If there's no body and only a commit message and signature line, we don't enter the `else` statement (since `len(msg_list == 2`). In this case, `msg_list` would still contain the subject and signature.